controllers/pipeline/pipeline_adapter.go

```diff
@@ -278,6 +301,26 @@ func (a *Adapter) getImagePullSpecFromSnapshotComponent(snapshot *applicationapi
 	return "", fmt.Errorf("couldn't find the requested component info in the given Snapshot")
 }
 
+// getComponentSourceFromSnapshotComponent gets the component source from the given Snapshot for the given Component,
+func (a *Adapter) getComponentSourceFromSnapshotComponent(snapshot *applicationapiv1alpha1.Snapshot, component *applicationapiv1alpha1.Component) (*applicationapiv1alpha1.ComponentSource, error) {
+	for _, snapshotComponent := range snapshot.Spec.Components {
+		if snapshotComponent.Name == component.Name {
+			return &snapshotComponent.Source, nil
+		}
+	}
+	return nil, fmt.Errorf("couldn't find the requested component source info in the given Snapshot")
+}
+
+// getComponentSourceFromComponent gets the component source from the given Component as Revision
+// and set Component.Status.LastBuiltCommit as Component.Source.GitSource.Revision if it is defined.
+func (a *Adapter) getComponentSourceFromComponent(component *applicationapiv1alpha1.Component) *applicationapiv1alpha1.ComponentSource {
+	componentSource := component.Spec.Source.DeepCopy()
+	if component.Status.LastBuiltCommit != "" {
+		componentSource.GitSource.Revision = component.Status.LastBuiltCommit
+	}
+	return componentSource
```

Comment:
@dirgim Should we return an error here when Revision is not found in componentSource?

Reply:
Not completely sure, the [Revision](https://github.com/redhat-appstudio/application-api/blob/main/api/v1alpha1/component_types.go#L39-L42) field is optional. My gut feeling is that we shouldn't return an error since the `componentSource` is still valid even without that field.

Reply:
Resolving this discussion for now so it doesn't get in the way of the review process.